pages/boards.tsx

```diff
@@ -0,0 +1,304 @@
+import styled from "styled-components";
+import React, { useEffect, useState } from "react";
+import Header from "@/components/Header";
+import BestArticle from "@/components/BestArticle";
+import axios from "@/lib/axios";
+import Image from "next/image";
+import search from "@/images/ic_search.png";
+import ArrowDown from "@/images/arrow_down.svg";
+import Article from "@/components/Article";
+import sort from "@/images/ic_sort.png";
+
+interface Article {
+  id: number;
+  title: string;
+  content: string;
+  image: string | null;
+  likeCount: number;
+  createdAt: Date;
+  updatedAt: Date;
+  writer: {
+    id: number;
+    nickname: string;
+  };
+}
+interface OptionsContainerProps {
+  isOpen: boolean;
+}
+
+export default function Boards() {
+  const [articles, setArticles] = useState<Article[] | undefined>(undefined);
+  const [isOpen, setIsOpen] = useState(false);
+  const [order, setOrder] = useState("recent");
```

Comment:
```suggestion
  const [order, setOrder] = useState<"recent" | "like">("recent");
```
위처럼 타입을 조금 더 좁혀주면 좋을 듯 합니다.